components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -1,64 +1,148 @@
+            break;
+        }
+        default: {
+            stateClassName += ' bg-kumquat text-red-600'
             break;
         }
     }
     let changesClassName = '';
     const pendingChanges = getPendingChanges(desc.latestInstance);
-    if (pendingChanges.length > 0) {
-        changesClassName += ' text-yellow-700';
+    const numberOfChanges = pendingChanges.reduceRight((i, c) => i + c.items.length, 0)
+    if (numberOfChanges > 0) {
+        changesClassName += 'text-sm text-yellow-600';
+    } else {
+        changesClassName += 'text-sm text-gray-400';
+    }
+    let changesLabel = 'No Changes';
+    if (numberOfChanges === 1) {
+        changesLabel = '1 Change';
+    } else if (numberOfChanges > 1) {
+        changesLabel = numberOfChanges + ' Changes';
     }
     const currentBranch = desc.latestInstance?.status.repo?.branch || Workspace.getBranchName(desc.workspace) || '<unknown>';
     const ws = desc.workspace;
-    return <div className="whitespace-nowrap flex space-x-2 py-8 border-b-2 border-gray-100 text-gray-800 w-full justify-between hover:bg-gray-100">
-        <div className={stateClassName}>
-            &nbsp;
+    const startUrl = new GitpodHostUrl(window.location.href).with({
+        pathname: '/start/',
+        hash: '#' + ws.id
+    }).toString();
+    const downloadURL = new GitpodHostUrl(window.location.href).with({ 
+        pathname: `/workspace-download/get/${ws.id}` 
+    }).toString();
+    const menuEntries: ContextMenuEntry[] = [
+        {
+            title: 'Open',
+            href: startUrl
+        },
+        {
+            title: 'Download',
+            href: downloadURL
+        },
+        {
+            title: 'Share',
+            active: !!ws.shareable,
+            onClick: () => {
+                gitpodService.server.controlAdmission(ws.id, ws.shareable ? "owner" : "everyone");
+            }
+        },
+        {
+            title: 'Pin',
+            active: !!ws.pinned,
+            separator: true,
+            onClick: () => {
+                gitpodService.server.updateWorkspaceUserPin(ws.id, 'toggle')
+            }
+        },
+        {
+            title: 'Delete',
+            customFontStyle: 'text-red-600',
+            onClick: () => {
+                setModalVisible(true);
+            }
+        }
+    ];
+    const project = getProject(ws);
+    return <div className="whitespace-nowrap flex space-x-2 py-6 w-full justify-between hover:bg-gray-100 rounded-lg">
+        <div className="w-10 self-center pl-4" >
+            <div className={stateClassName}>
+                &nbsp;
+            </div>
         </div>
         <div className="flex flex-col w-3/12">
-            <div className="text-gray-900">{ws.id}</div>
-            <div className="text-sm overflow-ellipsis truncate text-gray-400">{getProject(ws)}</div>
+            <Link to={"/start/#"+ws.id}><div className="font-medium text-gray-800 truncate hover:underline">{ws.id}</div></Link>
```

Comment:
Maybe re-use `startUrl`?

```suggestion
            <Link to={startUrl}><div className="font-medium text-gray-800 truncate hover:underline">{ws.id}</div></Link>
```